client/state/sites/plans/selectors/calculate-plan-upgrade-credits.ts

```diff
@@ -0,0 +1,38 @@
+import { getPlanDiscountedRawPrice } from './get-plan-discounted-raw-price';
+import { getSitePlanRawPrice } from './get-site-plan-raw-price';
+import isPlanAvailableForPurchase from './is-plan-available-for-purchase';
+import type { IAppState } from 'calypso/state/types';
+
+/**
+ * Calculate available plan credits given a set of displayed plans
+ *
+ * @param {*} state app state
+ * @param {*} siteId the selected site id
+ * @param {string[]}  consideredPlans Plans that are considered for the given calcuation
+ * @returns {number} The amount of credits available for the given plans
+ */
+export function calculatePlanUpgradeCredits(
+	state: IAppState,
+	siteId: number | undefined,
+	consideredPlans: string[]
+): number {
+	if ( ! siteId ) {
+		return 0;
+	}
+
+	const creditsPerPlan = consideredPlans.map( ( planName ) => {
+		const availableForPurchase = isPlanAvailableForPurchase( state, siteId, planName );
+		if ( ! availableForPurchase ) {
+			return 0;
+		}
+		const annualDiscountPrice = getPlanDiscountedRawPrice( state, siteId, planName );
+		const annualRawPrice = getSitePlanRawPrice( state, siteId, planName );
```

Comment:
Are you talking about this selector? (I couldn't find any other selector)
I think that selector might need to be renamed to `getSitePlanPrices` because it gets the price information for the current site plan, not any plan that we have. 

https://github.com/Automattic/wp-calypso/blob/f8f12f8d5aedb387ef12f466fde5b616d20c251b/client/state/plans/selectors/get-plan-prices.ts#L14-L26

Reply:
> I think that selector might need to be renamed to getSitePlanPrices because it gets the price information for the current site plan, not any plan that we have.

I don't think that's the case? It's a wrapper to the three/four main pricing selectors used for for plans (any plan). It will indeed compute a discounted price if it's also the site plan, but I think you can trust the raw price to be the correct price needed at places (as far as I recall, and unless something hasn't changed already).

Reply:
Sorry, I just read the comment and on the function call and assumed that is what it did. 

Thanks [I tried to incorporate this](https://github.com/Automattic/wp-calypso/pull/76100/commits/9add02615948db9134f997671b443d97ed4a491c).

However for some reasons the values that I get do not tally with production credits values for some reasons. I did some investigation and could not figure it out right now. So I thought of leaving this out for now to be investigated later.

Reply:
Interesting. I think it's best if we try and go through this selector as it returns values from both [getPlanDiscountedRawPrice](https://github.com/Automattic/wp-calypso/blob/261706c3c01db16bb91c619728d2a69ec36953c8/client/state/sites/plans/selectors/get-plan-discounted-raw-price.ts), [getDiscountedRawPrice](https://github.com/Automattic/wp-calypso/blob/261706c3c01db16bb91c619728d2a69ec36953c8/client/state/plans/selectors/get-discounted-raw-price.ts) to use in comparison. That was the idea pretty much, to unify the pricing.

Maybe @aneeshd16 can chime in? I believe I've seen some recent work on an update to one of the underlying selectors.

Reply:
Hi @jdc91, the commit you shared cannot be accessed. Can you please share the commit in a different branch so that we help debug?

Reply:
I have moved the commit branched off from this PR at https://github.com/Automattic/wp-calypso/pull/76361

Reply:
Thanks! I'll have a look at this.